rules/55-zz-scsi_id.rules

```diff
@@ -12,6 +12,15 @@ KERNEL=="sd*[!0-9]|sr*", \
 KERNEL=="cciss*", ENV{DEVTYPE}=="disk", \
 	IMPORT{program}="scsi_id --export --whitelisted -d $devnode", \
         ENV{.ID_BUS}="cciss"
+KERNEL=="st*[0-9]|nst*[0-9]", KERNELS=="[0-9]*:*[0-9]", \
+	ENV{.BSG_DEV}="$root/bsg/$id"
+KERNEL=="st*[0-9]|nst*[0-9]", KERNELS=="[0-9]*:*[0-9]", \
+	IMPORT{program}="scsi_id --whitelisted --export --device=$env{.BSG_DEV}", \
```

Comment:
Do we really need this `.BSG_DEV` temporary variable ?

Reply:
> Do we really need this `.BSG_DEV` temporary variable ?

Yes, because `IMPORT` doesn't support string substitutions. See **udev(7)** again:

> The NAME, SYMLINK, PROGRAM, OWNER, GROUP, MODE, SECLABEL, and RUN fields support simple string substitutions

For whatever reason, `IMPORT` is not in the list.

Reply:
Ok but then what's the point to split this into 2 separate rules ?

Reply:
I believe I tested merging the two rules into one and it didn't work. IIRC it turned out that the environment variable set with `ENV{.BSG_DEV}` was only available in the environment in the following rule, not while the rule itself was executed. The man page doesn't say anything about this. The way it is now, we're on the safe side, perhaps with a tiny performance cost.

Reply:
This comment was more about readability than performance cost.